python/amici/sbml_import.py

```diff
@@ -1473,6 +1468,40 @@ def _parse_special_functions(sym: sp.Expr, toplevel: bool = True) -> sp.Expr:
     return sym
 
 
+def _parse_piecewise_to_heaviside(args: Iterable[sp.Expr]) -> sp.Expr:
+    """
+   Piecewise functions cannot be transformed into C++ right away, but AMICI has
+   a special interface for Heaviside function, so we transform them
+
+    :param args:
+        symbolic expressions for arguments of the piecewise function
+    """
+    # how many condition-expression pairs will we have?
+    formula = sp.sympify('0')
+    lastroot = sp.sympify('1')
+    conditions = list(grouper(args, 2, False))
+    for coeff, root in conditions:
+        if root != sp.false:
+            # Sympy doesn't assume by default that the expression is real...
+            root._assumptions['extended_real'] = True
+            # we now need to convert the relational >, >=, ... expression into
+            # a trigger function which will be a Heaviside argument
+            if root.is_Relational:
+                trigger = root.args[0] - root.args[1]
+                if type(root) in (sp.core.relational.StrictLessThan,
+                                  sp.core.relational.LessThan):
+                    trigger = -trigger
+            else:
+                raise SBMLException('AMICI can not parse peicewise functions '
+                                    'with non-relational arguments.')
+            tmp = sp.Heaviside(trigger)
+            formula += coeff * tmp
+            lastroot -= tmp
```

Comment:
Not sure whether this formula is actually correct. https://docs.sympy.org/latest/modules/functions/elementary.html#piecewise states that **only the first** expression that is true will be returned. I think applying `sp.piecewise_fold` and then translating the generated `ITE` expression may have a better mileage.

don't like the naming `lastroot` here, can we have something more descriptive?

Reply:
Eh... won't work. Piecewise_fold goes the other way round. It converts ITE into piecewise, not away from it...
I am writing a parser here... yay,,,

Reply:
Hmm, I thought piecewise_fold goes both ways:

https://docs.sympy.org/latest/modules/functions/elementary.html